pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -577,6 +735,39 @@ func (r *hibernationReconciler) nodesReady(cd *hivev1.ClusterDeployment, remoteC
 	return true, nil
 }
 
+func (r *hibernationReconciler) operatorsReady(remoteClient client.Client, logger log.FieldLogger) (bool, error) {
+	logger.Debug("Checking if ClusterOperators are ready")
+	coList := &configv1.ClusterOperatorList{}
+	err := remoteClient.List(context.TODO(), coList)
+	if err != nil {
+		logger.WithError(err).Log(controllerutils.LogLevel(err), "Failed to fetch ClusterOperators")
+		err = errors.Wrap(err, "failed to fetch ClusterOperators")
+		return false, err
+	}
+	success := true
+
+	for _, co := range coList.Items {
+		for _, cosc := range co.Status.Conditions {
+			if cosc.Type == "Disabled" && cosc.Status == "True" {
+				continue
+			}
+
+			// Check that ClusterOperators are in a good state before we consider a cluster ready:
+			if (cosc.Type == "Available" && cosc.Status == configv1.ConditionFalse) ||
+				(cosc.Type == "Progressing" && cosc.Status == configv1.ConditionTrue) ||
+				(cosc.Type == "Degraded" && cosc.Status == configv1.ConditionTrue) {
+				logger.WithFields(log.Fields{
+					"clusterOperator": co.Name,
+					"condition":       cosc.Type,
+					"status":          cosc.Status,
+				}).Info("ClusterOperator is in undesired state")
+				success = false
+			}
+		}
+	}
+	return success, nil
+}
+
 func (r *hibernationReconciler) checkCSRs(cd *hivev1.ClusterDeployment, remoteClient client.Client, logger log.FieldLogger) (reconcile.Result, error) {
 	kubeClient, err := r.remoteClientBuilder(cd).BuildKubeClient()
 	if err != nil {
```

Comment:
(Latent) At the end of this function, we requeue with a 30s delay even if we found all the CSRs are approved. We should fix that.